### PR TITLE
Site traversal playwright tests

### LINKED ISSUE #1333: convert site traversal tests from Protractor to Playwright tests
For reference: https://docs.google.com/spreadsheets/d/1w9wjfm-02QHLhv0ObPAKKlqnDWeag379GIXwOcRU4W0/edit#gid=0

TODO:

- [ ] Write first test
- [ ] Create first page object
- [ ] Write remaining site traversal tests and page objects
- [ ] Add `npx playwright test` to GHA

### PR DESCRIPTION
**Pull request by JeanneSon**
## Description

The following tests have been converted to playwright

Bellows E2E Page Traversal
✓ Explore signup page
✓ Explore forgot password page
✓ Explore login page
✓ Explore change password page
✓ Explore activity page
✓ Explore project page
✓ Explore site admin page
✓ Explore user profile page

For details/remarks regarding converting the above tests, refer to the spreadsheet; https://docs.google.com/spreadsheets/d/1w9wjfm-02QHLhv0ObPAKKlqnDWeag379GIXwOcRU4W0/edit#gid=0

Fixes #1333 

### Type of Change

- E2E test

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have reviewed the title/description of this PR which will be used as the squashed PR commit message
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by megahirt — APPROVED** [review]
LGTM Well done on your first PR @JeanneSon !  Some good work in here!

File: test/e2e/pages/projects.page.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  readonly projectNames: Locator;
+  readonly createButton: Locator;
+  static readonly url: string = '/app/projects';
+
+  constructor(page: Page) {
+    this.page = page;
+    this.pageName = page.locator('.page-name >> text=My Projects');
+    this.createButton = page.locator('button:has-text("Start or Join a New Project")');
+    this.projectsList = page.locator('[data-ng-repeat="project in visibleProjects"]');
+    this.projectNames = this.projectsList.locator('a[href^="/app/lexicon"]');
```
[the remark this answers is not in the document]
Comment (megahirt) [reply test/e2e/pages/projects.page.ts:16]: In this case `^=` means the attribute value **starts with** `/app/lexicon`

File: test/e2e/pages/projects.page.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  }
+
+  async goto() {
+    await this.page.goto(ProjectsPage.url);
+    await expect(this.pageName).toBeVisible();
+  }
+
+  // TODO: write feature request: implement a waiting spinning somthing indicator - create github issue as feature request
```
[the remark this answers is not in the document]
Comment (JeanneSon) [reply test/e2e/pages/projects.page.ts:24]: the language picker can sometimes be slow in showing the results

File: test/e2e/site-traversal.spec.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  test('Explore login page', async ({ page }) => {
+    const loginPage = new LoginPage(page);
+    await loginPage.goto();
+
+    await loginPage.usernameInput.type('');
+    await loginPage.passwordInput.type('');
+    await loginPage.submitButton.click();
+  });
+
+  test('Explore change passsword page (admin)', async ({ adminTab }) => {
```
[the remark this answers is not in the document]
Comment (megahirt) [reply test/e2e/site-traversal.spec.ts:46]: Per our conversation in the office, we settled on "tab" and "adminTab" for the browser instantiation or session (what Playwright calls a "page").  Our use of the word "page" refers to an application view and it's associated page object model for E2E testing.  A future PR will refactor the variable naming.

File: test/e2e/pages/projects.page.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (megahirt) [reply test/e2e/pages/projects.page.ts:24]: thank you.

**Pull request merged**